src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetVersionsControllerTests.cs

```diff
@@ -574,17 +581,129 @@ public async Task DataSetDoesNotExist_Returns404()
+                .WithPreviewTokens(() => [DataFixture.DefaultPreviewToken()]);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+
+            var response = await GetDataSetVersion(
+                dataSetId: dataSet.Id,
+                dataSetVersion: dataSetVersion.PublicVersion,
+                previewTokenId: dataSetVersion.PreviewTokens[0].Id);
+
+            var viewModel = response.AssertOk<DataSetVersionViewModel>(useSystemJson: true);
+            Assert.NotNull(viewModel);
+        }
+
+        [Fact]
+        public async Task PreviewTokenIsExpired_Returns403()
+        {
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion(filters: 1, indicators: 1, locations: 1, timePeriods: 2)
+                .WithStatus(DataSetVersionStatus.Draft)
+                .WithDataSetId(dataSet.Id)
+                .WithPreviewTokens(() => [DataFixture.DefaultPreviewToken(expired: true)]);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+
+            var response = await GetDataSetVersion(
+                dataSetId: dataSet.Id,
+                dataSetVersion: dataSetVersion.PublicVersion,
+                previewTokenId: dataSetVersion.PreviewTokens[0].Id);
+
+            response.AssertForbidden();
+        }
+
+        [Fact]
+        public async Task PreviewTokenIsForWrongDataSetVersion_Returns403()
+        {
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            var (dataSetVersion1, dataSetVersion2) = DataFixture
+                .DefaultDataSetVersion(filters: 1, indicators: 1, locations: 1, timePeriods: 2)
+                .WithStatus(DataSetVersionStatus.Draft)
+                .WithDataSetId(dataSet.Id)
+                .WithPreviewTokens(() => [DataFixture.DefaultPreviewToken()])
+                .Generate(2)
+                .ToTuple2();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.AddRange(dataSetVersion1, dataSetVersion2));
+
+            var response = await GetDataSetVersion(
+                dataSetId: dataSet.Id,
+                dataSetVersion: dataSetVersion1.PublicVersion,
+                previewTokenId: dataSetVersion2.PreviewTokens[0].Id);
+
+            response.AssertForbidden();
+        }
+
+        [Theory]
+        [InlineData(DataSetVersionStatus.Processing)]
+        [InlineData(DataSetVersionStatus.Failed)]
+        [InlineData(DataSetVersionStatus.Mapping)]
```

Comment:
We've also got a `Cancelled` status that we should also cover.

To make it easier to add more statuses in the future, we might want to refactor out some re-usable theory data that can be used across all our different theories.

We have a bunch of different test suites that could re-use this theory data (including the authorization handler tests), so we could add a new `TheoryData` namespace and add the theory data there e.g.

```cs
public static class DataSetVersionStatusTheoryData
{
    public static readonly TheoryData<DataSetVersionStatus> UnavailableStatuses = new()
    {
        DataSetVersionStatus.Processing,
        DataSetVersionStatus.Failed,
        DataSetVersionStatus.Mapping,
        DataSetVersionStatus.Withdrawn,
        DataSetVersionStatus.Cancelled,
    };
}
```

It might be worth considering implementing it using `ClassData` as doing this with  `MemberData` is a bit clunkier. I'd be fine with either approach.

It should be noted that we have a bunch of other test suites outside the scope of this ticket that are also doing similar theory tests. Might be a good idea to tidy these up as well whilst we're here.